### ISSUE TITLE
/proc/<pid>/maps: Pathnames are '/' inside a container using kernel 5.0.0-27-generic

### ISSUE THREAD

**Issue by MichaelSimons** [moby/moby#39875]
Using kernel 5.0.0-27-generic with an Ubuntu 18.04 container, every pathname in /proc/<pid>/maps is / instead of the mapped file path.

**Steps to reproduce the issue:**
1. In an environment using kernel 5.0.0-25-generic (or earlier) 
$ sudo docker run ubuntu cat /proc/1/maps
56324a3f5000-56324a3fd000 r-xp 00000000 fc:01 665673 /bin/cat
56324a5fc000-56324a5fd000 r--p 00007000 fc:01 665673 /bin/cat
56324a5fd000-56324a5fe000 rw-p 00008000 fc:01 665673 /bin/cat
56324a8eb000-56324a90c000 rw-p 00000000 00:00 0 [heap]
7f8ab5f91000-7f8ab6178000 r-xp 00000000 fc:01 666073 /lib/x86_64-linux-gnu/libc-2.27.so
7f8ab6178000-7f8ab6378000 ---p 001e7000 fc:01 666073 /lib/x86_64-linux-gnu/libc-2.27.so
7f8ab6378000-7f8ab637c000 r--p 001e7000 fc:01 666073 /lib/x86_64-linux-gnu/libc-2.27.so
7f8ab637c000-7f8ab637e000 rw-p 001eb000 fc:01 666073 /lib/x86_64-linux-gnu/libc-2.27.so
7f8ab637e000-7f8ab6382000 rw-p 00000000 00:00 0
7f8ab6382000-7f8ab63a9000 r-xp 00000000 fc:01 666055 /lib/x86_64-linux-gnu/ld-2.27.so
7f8ab6583000-7f8ab65a7000 rw-p 00000000 00:00 0
7f8ab65a9000-7f8ab65aa000 r--p 00027000 fc:01 666055 /lib/x86_64-linux-gnu/ld-2.27.so
7f8ab65aa000-7f8ab65ab000 rw-p 00028000 fc:01 666055 /lib/x86_64-linux-gnu/ld-2.27.so
7f8ab65ab000-7f8ab65ac000 rw-p 00000000 00:00 0
7ffc25954000-7ffc25975000 rw-p 00000000 00:00 0 [stack]
7ffc259f1000-7ffc259f4000 r--p 00000000 00:00 0 [vvar]
7ffc259f4000-7ffc259f5000 r-xp 00000000 00:00 0 [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0 [vsyscall]

2. In an environment using kernel 5.0.0-27-generic (broken):
$ sudo docker run ubuntu cat /proc/1/maps
55de0e052000-55de0e05a000 r-xp 00000000 fc:01 665673 /
55de0e259000-55de0e25a000 r--p 00007000 fc:01 665673 /
55de0e25a000-55de0e25b000 rw-p 00008000 fc:01 665673 /
55de0fcb3000-55de0fcd4000 rw-p 00000000 00:00 0 [heap]
7f7a8d881000-7f7a8da68000 r-xp 00000000 fc:01 666073 /
7f7a8da68000-7f7a8dc68000 ---p 001e7000 fc:01 666073 /
7f7a8dc68000-7f7a8dc6c000 r--p 001e7000 fc:01 666073 /
7f7a8dc6c000-7f7a8dc6e000 rw-p 001eb000 fc:01 666073 /
7f7a8dc6e000-7f7a8dc72000 rw-p 00000000 00:00 0
7f7a8dc72000-7f7a8dc99000 r-xp 00000000 fc:01 666055 /
7f7a8de73000-7f7a8de97000 rw-p 00000000 00:00 0
7f7a8de99000-7f7a8de9a000 r--p 00027000 fc:01 666055 /
7f7a8de9a000-7f7a8de9b000 rw-p 00028000 fc:01 666055 /
7f7a8de9b000-7f7a8de9c000 rw-p 00000000 00:00 0
7ffc744bd000-7ffc744de000 rw-p 00000000 00:00 0 [stack]
7ffc7452d000-7ffc74530000 r--p 00000000 00:00 0 [vvar]
7ffc74530000-7ffc74531000 r-xp 00000000 00:00 0 [vdso]

**Describe the results you received:**
Every pathname in /proc/<pid>/maps is / instead of the mapped file path

**Describe the results you expected:**
Every pathname in /proc/<pid>/maps is / the mapped file path

**Additional information you deem important (e.g. issue happens only occasionally):**

There is a kernel issue for this as well -https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1843018.  I wanted to log a Docker issue for visibility in case others are affected by it and to verify Docker isn't at fault here.

This is breaking .NET Core apps - https://github.com/dotnet/corefx/issues/40850

The problem can be reproduced in various Ubuntu releases, e.g. 18.04 and 19.04.

**Output of `docker version`:**

```
Client: Docker Engine - Community                           
 Version:           19.03.2                                 
 API version:       1.40                                    
 Go version:        go1.12.8                                
 Git commit:        6a30dfc                                 
 Built:             Thu Aug 29 05:29:11 2019                
 OS/Arch:           linux/amd64                             
 Experimental:      false                                   
                                                            
Server: Docker Engine - Community                           
 Engine:                                                    
  Version:          19.03.2                                 
  API version:      1.40 (minimum version 1.12)             
  Go version:       go1.12.8                                
  Git commit:       6a30dfc                                 
  Built:            Thu Aug 29 05:27:45 2019                
  OS/Arch:          linux/amd64                             
  Experimental:     false                                   
 containerd:                                                
  Version:          1.2.6                                   
  GitCommit:        894b81a4b802e4eb2a91d1ce216b8817763c29fb
 runc:                                                      
  Version:          1.0.0-rc8                               
  GitCommit:        425e105d5a03fabd737a126ad93d62a9eeede87f
 docker-init:                                               
  Version:          0.18.0                                  
  GitCommit:        fec3683     
```

**Output of `docker info`:**

```
Client:
 Debug Mode: false

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 10
 Server Version: 19.03.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 894b81a4b802e4eb2a91d1ce216b8817763c29fb
 runc version: 425e105d5a03fabd737a126ad93d62a9eeede87f
 init version: fec3683
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.0.0-1018-azure
 Operating System: Ubuntu 18.04.3 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 13.67GiB
 Name: mikeharder-docker-test
 ID: L2CI:XKAM:WL3O:UUFW:T3EQ:QLCH:CGOR:7U4A:ORCL:JXFY:7VRN:ERQ7
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support
```

**Comment by thaJeztah** [comment]
thanks for reporting!

/cc @crosbymichael @kolyshkin @sforshee FYI

**Comment by sforshee** [comment]
@thaJeztah thanks, we've also had other reports of this issue. I'm currently looking into it.

**Comment by kolyshkin** [comment]
Seems to be a kernel bug

**Comment by Josh-G** [comment]
this was fixed in kernel 5.0.0-31.33

**Comment by thaJeztah** [comment]
@josh-g thanks for the update; looks like this is the patch; https://lists.ubuntu.com/archives/kernel-team/2019-September/103893.html


I'll close this issue, but feel free to continue the conversation 👍
